src/librustc/ty/layout.rs

```diff
@@ -546,7 +546,8 @@ pub struct Struct {
 
     pub min_size: Size,
 
-    /// Size of padding needed by the struct due to requested repr alignment
+    /// Size of padding needed by the struct due to requested repr alignment.
+    /// This is created as required and will be empty if the structs fields require no padding.
     pub padding: Vec<Size>,
```

Comment:
Ok, so if emit all padding we do that we can make `memory_index_with_padding_fields` always `O(1)` (basically index * 2) and calculate padding on demand.